Updating tf.experimental.numpy.vander for N=0

SuryanarayanaY opened the pull request:
At present the API `tf.experimental.numpy.vander` behaves differently compared to its Numpy variant `numpy.vander` when the argument value of `N=0` .

I think the behaviour of Numpy variant is correct.Current TF implementation broadcasts N to `N=shape(x)[0]` when either N=0 or N=None which is not correct. When N=0 there should not be any broadcasting which is equivalent to `numpy.vander` behaviour.Hence proposing the changes in code to get numpy behaviour.

Attached the gist for demo in same with and without the code changes and with the proposed change in code numpy behaviour can be achievable.

Fixes #60827

SuryanarayanaY committed:
Updating tf.experimental.numpy.vander for N=0

At present the API tf.experimental.numpy.vander behaves differently compared to its Numpy variant numpy.vander when the argument value of N=0 .

I think the behaviour of Numpy variant is correct.Current TF implementation broadcasts N to N=shape(x)[0] when either N=0 or N=None which is not correct. When N=0 there should not be any broadcasting which is numpy behaviour.Hence proposing the changes in code to get numpy behaviour.

cantonios commented on tensorflow/python/ops/numpy_ops/np_array_ops.py:
```diff
@@ -1367,8 +1367,9 @@ def vander(x, N=None, increasing=False):  # pylint: disable=missing-docstring,in
   x_shape = array_ops.shape(x)
-  N = N or x_shape[0]
-
+  if N != 0:
```
For better clarity you should probably just use the simple if statement:
```
if N is None:
  N = x_shape[0]
```

cantonios reviewed (changes requested):
Can you add a test to verify the new behavior?

gbaned commented:
Hi @SuryanarayanaY Can you please check @cantonios's [comments](https://github.com/tensorflow/tensorflow/pull/61079#pullrequestreview-1498899933) ? Thank you!

SuryanarayanaY committed:
Update np_array_ops.py

Updated the code as requested

SuryanarayanaY commented:
Changes done as requested. Thanks!

cantonios reviewed (changes requested):
Still awaiting a test for the new behavior

gbaned commented:
Hi @SuryanarayanaY Any update on this PR? Please. Thank you!

SuryanarayanaY committed:
Added test case for Changes in tf.experimental.numpy.vander

Added a test case for checking the behaviour wrt numpy for the changes done in tf.experimental.numpy.vander.

SuryanarayanaY committed:
Resolved indentation errors in np_array_ops.py

Resolved indentation error in vander function
